doc/mgmt/SONiC_IntfRangeFeature_HLD.md

```diff
@@ -0,0 +1,235 @@
+* Extend show and delete interface commands to support range of interfaces.
+
+### 1.1.3 Scalability Requirements
+NA
+### 1.1.4 Warm Boot Requirements
+NA
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+- KLISH XML changes for adding new CLI view for interface range command.
+- Use of VAR tag in KLISH XML to cache existing interfaces list.
+- Changes in KLISH actioner script to invoke APIs for range of interfaces.
+- Use **interface range** command to configure existing interfaces in the range.
+- Use **interface range create** command to create non-existing interfaces in the range.
+
+### 1.2.2 Container
+This feature is implemented within the Management Framework container.
+
+### 1.2.3 SAI Overview
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+## 2.2 Functional Description
+# 3 Design
+## 3.1 Overview
+* Creation and configuration of interfaces to be done as a single transaction.
+* Transaction for show and delete of interfaces will be per interface.
+* Modifying XML file to use correct PTYPE to support number, range, or comma-delimited list of numbers and ranges.
+Example: `ETHER_INTERFACE_RANGE` for physical interfaces<br>
+```
+<PARAM
+    name="phy-if-name"
+    help="Physical interface range"
+    ptype="ETHER_INTERFACE_RANGE"
+/>
+```
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+## 3.4 SyncD
+## 3.5 SAI
+## 3.6 User Interface
+### 3.6.1 Data Models
+NA
+### 3.6.2 CLI
+#### 3.6.2.1 Interface range commands in native mode
+
+####  3.6.2.1.1 Ethernet interface range:
+##### Configure physical interfaces in given range
+Syntax: `interface range Ethernet port - port[,port - port...]`
+```
+sonic(config)# interface range ?
+Ethernet         Physical interface range
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+create           Create interfaces in range
+
+sonic(config)# interface range Ethernet 1-4,5
+sonic(conf-if-range-Eth1-4,5)# ?
+channel-group   Configure PortChannel parameters
+end             Exit to the exec Mode
+exit            Exit from current mode
+link            Interface link                   
+mac             Configure MAC parameters
+mtu             Configure MTU        
+nat-zone        NAT Zone               
+no              Negate a command or set its defaults
+service-policy  Apply ingress or egress policy      
+sflow           Interface sFlow settings
+shutdown        Disable the interface     
+spanning-tree   Configure spanning tree on interface      
+storm-control   configure storm-control   
+switchport      Configure switchport parameters      
+threshold       Configure priority-group/queue threshold on an interface.
+udld            UDLD interface configuration subcommands
+
+sonic(conf-if-range-Eth1-4,5)# mtu 9000
```

Comment:
what will be the behavior when one of the interface in the range has IP address configured and user tries enable spanning-tree or channel-group. Will the command be applied on all interface except the one which IP address. If yes, will there be any info message for the user.